### PR TITLE
feat(console): api proxy endpoints groups

### PR DESCRIPTION
**Pull request by Okhelifi**
## Issue

https://graviteecommunity.atlassian.net/browse/APIM-125

## Description

Migrate api proxy endpoint  groups pages

#### Read only

Screenshot 2022-10-18 at 12 12 31

Screenshot 2022-10-18 at 12 12 51

Screenshot 2022-10-18 at 12 13 08

#### Edit

Screenshot 2022-10-18 at 12 13 49

Screenshot 2022-10-18 at 12 14 12

Screenshot 2022-10-18 at 12 14 28

#### Read only

Screenshot 2022-10-24 at 13 04 39

Screenshot 2022-10-24 at 13 05 03

#### Edit

Screenshot 2022-10-24 at 13 05 42

Screenshot 2022-10-24 at 13 06 55

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/feat-125-edit-endpoints-groups/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-vczrkeiugf.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/general/api-proxy-group-general.component.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        <mat-error *ngIf="generalForm.get('name').hasError('pattern')"> Group name does not support <code>:</code> in its name. </mat-error>
+        <mat-error *ngIf="generalForm.get('name').hasError('isUnique')"> This name is used by another endpoint group. </mat-error>
+      </mat-form-field>
+    </div>
+
+    <!-- Load balancing algorithm -->
+    <div class="card__group-lb">
+      <div class="card__group-lb__label">Load balancing algorithm</div>
+      <p class="card__group-lb__hint">
+        You can configure load balancing by creating a logical grouping of endpoints and specifying a load balancing algorithm for them.
```
Comment (a-cordier) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/general/api-proxy-group-general.component.html]:
```suggestion
        You can configure load balancing by creating a logical group of endpoints and specifying a load balancing algorithm for them.
```

Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/general/api-proxy-group-general.component.html]: Done

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<form *ngIf="!!serviceDiscoveryForm && !!serviceDiscoveryItems" [formGroup]="serviceDiscoveryForm" gioFormFocusInvalid>
+  <mat-card>
+    <div class="card__group-sd__enable">
+      <div class="card__group-sd__enable__label">Enabled service discovery</div>
+      <mat-checkbox aria-label="Enable service discovery" formControlName="enabled">
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.html]: I would have expected to have a toggle (with gio wrapper) like on the other screens, what do you think ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.html]: same question for the other checkboxes used to activate or not a feature :)

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/api-proxy-group-edit.component.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+          this.serviceDiscoveryItems = serviceDiscoveryItems;
+          this.initForms();
+        }),
+      )
+      .subscribe();
+
+    this.connectorService
+      .list(true)
+      .pipe(
+        map((connectors) => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/api-proxy-group-edit.component.ts:97]:
```suggestion
        takeUntil(this.unsubscribe$),
        map((connectors) => {
```

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      serviceDiscoveryValues: event.detail?.values,
+    });
+  }
+
+  private onFormValuesChange(type: string) {
+    if (type) {
+      this.serviceDiscoveryService
+        .getSchema(type)
+        .pipe(
+          map((schema) => {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.ts:82]: takeUntil(this.unsubscribe$), no ?

**Review by ThibaudAV — APPROVED** [review]
Audrey had remarked to me that it might be better to have a button with text here
image


I have the impression that there is a bug when you do a new endpoint -> configuration. nothing is displayed 🤔

**Comment by Okhelifi** [comment]
> Audrey had remarked to me that it might be better to have a button with text here image
> 
> I have the impression that there is a bug when you do a new endpoint -> configuration. nothing is displayed 🤔

You need first to select a type. But I can use Http by default.

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/api-proxy-group-edit.component.ts (hunk 3 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/api-proxy-group-edit.component.ts:97]: done

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.ts (hunk 4 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.ts:82]: done

File: gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.html (hunk 2 above)
Comment (Okhelifi) [reply gravitee-apim-console-webui/src/management/api/proxy/endpoints/groups/edit/service-discovery/api-proxy-group-service-discovery.component.html]: Done

**Pull request merged**